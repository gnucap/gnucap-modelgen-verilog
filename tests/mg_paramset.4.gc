
attach ./modelgen_1.so

verilog

`modelgen
module base(p, n);
  electrical p, n;
  (*desc="m"*) real m;
  analog begin
    m = $mfactor();
  end
endmodule
paramset test_ps4a base
endparamset
paramset test_ps4b base
endparamset
`endmodelgen

!make test_ps4b.so > /dev/null
attach ./test_ps4b.so

verilog

test_ps4a #() a(1, 0);
test_ps4b #(.$mfactor(4)) b(1, 0);

print dc m(a) m(b)
dc
status notime
