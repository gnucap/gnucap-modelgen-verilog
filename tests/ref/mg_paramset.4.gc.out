#           m(a)       m(b)      
 0.         1.         4.        
Gnucap   System status
iterations: op=0, dc=2, tran=0, fourier=0, total=2
transient timesteps: accepted=0, rejected=0, total=0
nodes: user=1, subckt=0, model=0, total=1
dctran density=100.0%, ac density=100.0%
